### PR TITLE
syntax_ext: format: nest_level's are no more

### PR DESCRIPTION
**Pull request by xen0n**
Just noticed this while working on #33642 and here's a quick fix, shouldn't touch anything else. It's some historic code indeed...

### CODE REVIEW

File: src/libsyntax_ext/format.rs [hunk 1]
```diff
@@ -68,7 +68,6 @@ struct Context<'a, 'b:'a> {
     name_positions: HashMap<String, usize>,
 
     /// Updated as arguments are consumed or methods are entered
```
Comment (sanxiyn) [inline src/libsyntax_ext/format.rs]: Should delete the comment too.

Comment (xen0n) [inline src/libsyntax_ext/format.rs]: Oops, sure. Note that `next_arg` is still used for verifying types for implicit positional args, so I'll only delete the reference to methods.

Comment (xen0n) [inline src/libsyntax_ext/format.rs]: Done, should I squash my commits or will it be merged as such?

**Pull request merged**